Apps/WebClient/src/Server/Services/AuthService.cs

```diff
@@ -39,12 +39,21 @@ public AuthService(ILogger<AuthService> logger, IHttpContextAccessor httpContext
         /// Authenticates the request based on the current context.
         /// </summary>
         /// <returns>The AuthData containing the token and user information.</returns>
-        public async Task<Models.AuthData> Authenticate()
+        public async Task<Models.AuthData> GetAuthenticationData()
         {
-            this.logger.LogTrace("Getting Bearer token");
-            var user = this.httpContextAccessor.HttpContext.User.FindFirst(ClaimTypes.NameIdentifier).Value;
-            var token = await this.httpContextAccessor.HttpContext.GetTokenAsync("access_token").ConfigureAwait(true);
-            return new Models.AuthData { IsAuthenticated = true, Token = token, User = user };
+            Models.AuthData authData = new Models.AuthData();
+            ClaimsPrincipal user = this.httpContextAccessor.HttpContext.User;
+            authData.IsAuthenticated = user.Identity.IsAuthenticated;
+            if (authData.IsAuthenticated)
+            {
+                this.logger.LogDebug("Getting Authentication data");
```

Comment:
We could instead use the constructor for User and pass that one into the constructor of AuthData.
```
User user = new User { 
       Id= user.FindFirstValue("preferred_username"), 
       Name = user.FindFirstValue("name"),
       Email = user.FindFirstValue(ClaimTypes.Email)
}
Models.AuthData authData = new Models.AuthData(user, user.Identity.IsAuthenticated);
```

That way the AuthData object is guaranteed to have a User if it exists, otherwise will null

Reply:
resolved